File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dict = {}
    for cloth, type in clothes:
        if type in dict.keys():
            dict[type] += [cloth]
        else:
            dict[type] = [cloth]
            
    for i, val in dict.items():
        answer *= (len(val)+1)
            
    return answer -1

File: 프로그래머스/2/42578. 의상/의상.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(string[,] clothes) {
        int answer = 1;
        Dictionary<string, int> types = new Dictionary<string, int>();
        for(int i=0; i<clothes.GetLength(0); i++){
            if(!types.TryAdd(clothes[i,1], 1)){
                types[clothes[i,1]]++;
            }
        }
        
        foreach(int value in types.Values){
            answer *= (value+1);
        }
        return answer-1;
    }
}